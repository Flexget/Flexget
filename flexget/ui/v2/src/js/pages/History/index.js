import { connect } from 'react-redux';
import { request } from 'utils/actions';
import { GET_HISTORY } from 'store/history/actions';
import History from './History';

function mapDispatchToProps(dispatch) {
  return {
    getHistory: data => dispatch(request(GET_HISTORY, data)),
  };
}

export default connect(null, mapDispatchToProps)(History);
export { History };
