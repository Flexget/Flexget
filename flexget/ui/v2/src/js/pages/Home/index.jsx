import React from 'react';
import InfoCard from './InfoCard';

const Home = () => <InfoCard />;

export default Home;
