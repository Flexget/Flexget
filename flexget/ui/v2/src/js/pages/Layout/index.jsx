import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Logo from 'pages/Layout/Logo';
import Navbar from 'pages/Layout/Navbar';
import SideNav from 'pages/Layout/SideNav';
import LoadingBar from 'common/LoadingBar';
import ErrorStatus from 'common/ErrorStatus';
import InfoStatus from 'common/InfoStatus';
import {
  Wrapper,
  Main,
  Header,
  LogoWrapper,
  Nav,
  SideBar,
  Content,
} from './styles';

class Layout extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
  };

  static defaultProps = {
    loggedIn: false,
  };

  state = {
    sideBarOpen: (window.matchMedia && !!window.matchMedia('(min-width: 600px)').matches) || false,
  };

  toggleSideBar = () => {
    this.setState({
      sideBarOpen: !this.state.sideBarOpen,
    });
  }

  render() {
    const { children } = this.props;
    const { sideBarOpen } = this.state;

    return (
      <Wrapper>
        <Header>
          <LogoWrapper>
            <Logo sideBarOpen={sideBarOpen} />
          </LogoWrapper>
          <Nav>
            <Navbar toggle={this.toggleSideBar} />
            <LoadingBar />
          </Nav>
        </Header>
        <Main>
          <SideBar>
            <SideNav
              sideBarOpen={sideBarOpen}
              toggle={this.toggleSideBar}
            />
          </SideBar>
          <Content open={sideBarOpen}>
            { children }
          </Content>
          <ErrorStatus />
          <InfoStatus />
        </Main>
      </Wrapper>
    );
  }
}

export default Layout;
