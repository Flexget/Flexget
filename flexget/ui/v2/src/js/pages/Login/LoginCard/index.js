import { connect } from 'react-redux';
import { LOGIN } from 'store/auth/actions';
import { request } from 'utils/actions';
import LoginCard from './LoginCard';

export function mapStateToProps({ status }) {
  return {
    initialValues: {
      username: 'flexget',
    },
    errorStatus: status.error || {},
  };
}

function mapDispatchToProps(dispatch) {
  return {
    onSubmit: credentials => dispatch(request(LOGIN, credentials)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(LoginCard);
export { LoginCard };
