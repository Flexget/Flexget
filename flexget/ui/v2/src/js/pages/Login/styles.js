import styled from 'react-emotion';
import theme from 'theme';
import headerImage from 'images/header.png';

export const Logo = styled.div`
  background: transparent url(${headerImage}) no-repeat center;
  min-height: 9rem;
  background-size: 100% auto;
  margin: 0 1rem;
  ${theme.breakpoints.up('sm')} {
    background-size: auto;
  }
`;
