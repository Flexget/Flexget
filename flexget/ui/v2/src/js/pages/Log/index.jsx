import React from 'react';
import { PaperWrapper } from 'common/styles';
import Header from './Header';
import LogTable from './LogTable';
import { TableWrapper } from './styles';

const LogPage = () => (
  <PaperWrapper elevation={4}>
    <Header />
    <TableWrapper>
      <LogTable />
    </TableWrapper>
  </PaperWrapper>
);

export default LogPage;
