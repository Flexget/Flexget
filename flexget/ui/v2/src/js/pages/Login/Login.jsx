import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import LoginCard from './LoginCard';
import SplashScreen from './Splash';
import { Logo } from './styles';

const LoginPage = ({ redirectToReferrer, location, loading }) => {
  const { from } = location.state || { from: { pathname: '/' } };

  if (redirectToReferrer) {
    return (
      <Redirect to={from} />
    );
  }

  if (loading) {
    return <SplashScreen />;
  }

  return (
    <div>
      <Logo />
      <LoginCard />
    </div>
  );
};

LoginPage.propTypes = {
  location: PropTypes.object.isRequired,
  redirectToReferrer: PropTypes.bool.isRequired,
  loading: PropTypes.bool.isRequired,
};

export default LoginPage;
