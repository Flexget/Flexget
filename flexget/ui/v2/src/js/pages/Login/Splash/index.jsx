import React from 'react';
import WaveSpinner from 'common/WaveSpinner';
import { Logo } from './styles';

const SplashScreen = () => (
  <div>
    <Logo />
    <WaveSpinner />
  </div>
);


export default SplashScreen;
