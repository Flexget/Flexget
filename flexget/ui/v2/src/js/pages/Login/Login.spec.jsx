import React from 'react';
import { Login, mapStateToProps } from 'pages/Login';
import renderer from 'react-test-renderer';
import { provider, router, themed } from 'utils/tests';

const Component = () => <div />;
describe('pages/Login', () => {
  describe('Login', () => {
    it('renders correctly when logged in', () => {
      const tree = renderer.create(
        router(themed(<Login
          component={Component}
          redirectToReferrer
          location={{}}
        />))
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders correctly when logged out', () => {
      const tree = renderer.create(
        provider(router(themed(<Login
          component={Component}
          redirectToReferrer={false}
          location={{}}
        />)), { status: {} })
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('should return logged in if logged in', () => {
      expect(mapStateToProps({
        auth: { loggedIn: true },
        status: { loading: {} },
      })).toMatchSnapshot();
    });

    it('should return not logged in if logged out', () => {
      expect(mapStateToProps({ auth: { }, status: { loading: {} } })).toMatchSnapshot();
    });
  });
});
