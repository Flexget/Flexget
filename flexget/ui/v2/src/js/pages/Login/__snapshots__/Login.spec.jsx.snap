// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Login Login renders correctly when logged in 1`] = `null`;

exports[`pages/Login Login renders correctly when logged out 1`] = `
.glamor-0 {
  background: transparent url(test-file-stub) no-repeat center;
  min-height: 9rem;
  background-size: 100% auto;
  margin: 0 1rem;
}

.glamor-6 {
  max-width: 40rem;
  margin: 0 1rem;
}

.glamor-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.glamor-1 {
  color: #f44336;
  text-align: center;
  padding: 1rem;
}

.glamor-2 {
  padding-bottom: 1rem;
}

.glamor-5 {
  width: 100%;
}

@media (min-width:600px) {
  .glamor-0 {
    background-size: auto;
  }
}

@media (min-width:600px) {
  .glamor-6 {
    margin: 0 auto;
  }
}

<div>
  <div
    className="glamor-0"
  />
  <div
    className="MuiPaper-root-1 MuiPaper-shadow2-5 MuiPaper-rounded-2 glamor-6"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="MuiCardContent-root-28 glamor-4"
      >
        <div
          className="glamor-1"
        />
        <div
          className="MuiFormControl-root-29 glamor-2"
          onBlur={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          required={false}
        >
          <label
            className="MuiFormLabel-root-39 MuiInputLabel-root-33 MuiInputLabel-formControl-34 MuiInputLabel-animated-37"
            data-shrink={false}
            htmlFor="username"
          >
            Username
          </label>
          <div
            className="MuiInput-root-43 MuiInput-formControl-44 MuiInput-inkbar-45 MuiInput-underline-51"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <input
              autoComplete={undefined}
              autoFocus={undefined}
              className="MuiInput-input-47 MuiInput-inputSingleline-54"
              defaultValue={undefined}
              disabled={false}
              id="username"
              name="username"
              onChange={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={undefined}
              rows={undefined}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="MuiFormControl-root-29 glamor-2"
          onBlur={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          required={false}
        >
          <label
            className="MuiFormLabel-root-39 MuiInputLabel-root-33 MuiInputLabel-formControl-34 MuiInputLabel-animated-37"
            data-shrink={false}
            htmlFor="password"
          >
            Password
          </label>
          <div
            className="MuiInput-root-43 MuiInput-formControl-44 MuiInput-inkbar-45 MuiInput-underline-51"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <input
              autoComplete={undefined}
              autoFocus={undefined}
              className="MuiInput-input-47 MuiInput-inputSingleline-54"
              defaultValue={undefined}
              disabled={false}
              id="password"
              name="password"
              onChange={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={undefined}
              rows={undefined}
              type="Password"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="MuiCardActions-root-58"
      >
        <button
          className="MuiButtonBase-root-74 MuiButton-root-60 MuiCardActions-actionSpacing-59 glamor-5"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label-62"
          >
            Login
          </span>
          <span
            className="MuiTouchRipple-root-76"
          />
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`pages/Login mapStateToProps should return logged in if logged in 1`] = `
Object {
  "loading": false,
  "redirectToReferrer": true,
}
`;

exports[`pages/Login mapStateToProps should return not logged in if logged out 1`] = `
Object {
  "loading": false,
  "redirectToReferrer": false,
}
`;
